interbotix_ros_xsarms/examples/interbotix_xsarm_gravity_compensation/launch/interbotix_gravity_compensation.launch.py

```diff
@@ -0,0 +1,132 @@
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from typing import List
+
+from launch import LaunchDescription
+from launch.actions import (
+    IncludeLaunchDescription,
+    DeclareLaunchArgument,
+    OpaqueFunction
+)
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import (
+    LaunchConfiguration,
+    PathJoinSubstitution,
+)
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+
+from interbotix_xs_modules.xs_common import get_interbotix_xsarm_models
+
+
+def launch_setup(context, *args, **kwargs):
+    # Define the launch arguments
+    robot_model_launch_arg = LaunchConfiguration('robot_model')
+    robot_name_launch_arg = LaunchConfiguration('robot_name')
+    motor_specs_launch_arg = LaunchConfiguration('motor_specs')
+
+    # Create the gravity compensation node
+    gravity_compensation_node = Node(
+        package='interbotix_gravity_compensation',
+        executable='interbotix_gravity_compensation',
+        name='gravity_compensation',
+        namespace=robot_name_launch_arg,
+        output='screen',
+        emulate_tty=True,
+        parameters=[{'motor_specs': motor_specs_launch_arg}]
+    )
+
+    # include the xsarm_control_launch with arguments
+    xsarm_control_launch = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource([
+            PathJoinSubstitution([
+                FindPackageShare('interbotix_xsarm_control'),
+                'launch',
+                'xsarm_control.launch.py'
+            ])
+        ]),
+        launch_arguments={
+            'robot_model': robot_model_launch_arg,
+            'robot_name': robot_name_launch_arg,
+        }.items()
+    )
+
+    return [gravity_compensation_node, xsarm_control_launch]
+
+
+def generate_launch_description():
+    # Create a list of all the launch arguments
+    declared_arguments: List[DeclareLaunchArgument] = []
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            'robot_model',
+            choices=get_interbotix_xsarm_models(),
+            description='model type of the Interbotix Arm such as `wx200` or `rx150`.'
+        )
+    )
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            'robot_name',
+            default_value=LaunchConfiguration('robot_model'),
+            description=(
+                'name of the robot (typically equal to `robot_model`, but could be anything).'
+            ),
+        )
+    )
+    declared_arguments.append(
+        DeclareLaunchArgument(
+            'motor_specs',
+            default_value=[
+                PathJoinSubstitution([
+                    FindPackageShare('interbotix_xsarm_gravity_compensation'),
+                    'config']),
+                "/motor_specs_",
```

Comment:
This is a bit messy. Try to do something like this: https://github.com/Interbotix/interbotix_ros_manipulators/blob/552368e2ac0f8fdba69ee792ad71381238d78118/interbotix_ros_xsarms/examples/interbotix_xsarm_puppet/launch/xsarm_puppet.launch.py#L228-L231

Reply:
Is this better?

```python
            default_value=[PathJoinSubstitution([
                FindPackageShare('interbotix_xsarm_gravity_compensation'),
                'config',
                "motor_specs_"]),
                LaunchConfiguration('robot_model'),
                '.yaml'
            ],
```

Reply:
Would prefer the style below:

```python
default_value=[PathJoinSubstitution([
    FindPackageShare('interbotix_xsarm_gravity_compensation'),
    'config',
    'motor_specs_']), LaunchConfiguration('robot_model'), '.yaml'
],
```

or

```python
default_value=PathJoinSubstitution([
    FindPackageShare('interbotix_xsarm_gravity_compensation'),
    'config',
    f'motor_specs_{LaunchConfiguration('robot_model').perform(context)}.yaml',
]),
```